### DISCUSSION TITLE
Output to GCS(Google Cloud Storage) using Service account not HMAC

### DISCUSSION THREAD

**Discussion by hwanghakbeom** [fluent/fluent-bit#6636] [Q&A]
Hello.  and sorry for my bad english first.

I recently Tested and got result logs to GCS using HMAC and S3 output plugin.

It was enough to send logs to google cloud platform

When I keep consistency with other google plugins BigQuery and Stackdriver, or manage secret key, Probably need feature what Logs to GCS using Google Service Account. 

Is it good idea to make a new feature? 

or Do I have to keep using S3 output plugins with Google HMAC key.

I am started to reading contribution guide to help to make this feature even I am afraid of c code because I believe this is open-source. 

if any plan or idea. please let me know.

**Comment by Z4ck404** [comment]
A gcp service account is limited to a maximum of 10 keys, and these keys remain valid indefinitely unless explicitly deleted. However, using these keys with Fluent Bit alongside the S3 plugin presents certain limitations.
In one of our use cases, we need to generate temporary sts credentials that Fluent Bit agents can use to collect and transmit data to cloud object storage. Implementing the logic for temporary credentials using HMAC keys can be complex, contrasting with AWS where this process is more straightforward. It would be greatly beneficial if Fluent Bit had an output plugin for GCS similar to AWS, streamlining the integration process and flexibility.

**Comment by RadhaYarra193737** [comment]
> Hello. and sorry for my bad english first.
> 
> I recently Tested and got result logs to GCS using HMAC and S3 output plugin.
> 
> It was enough to send logs to google cloud platform
> 
> When I keep consistency with other google plugins BigQuery and Stackdriver, or manage secret key, Probably need feature what Logs to GCS using Google Service Account.
> 
> Is it good idea to make a new feature?
> 
> or Do I have to keep using S3 output plugins with Google HMAC key.
> 
> I am started to reading contribution guide to help to make this feature even I am afraid of c code because I believe this is open-source.
> 
> if any plan or idea. please let me know.

Hi @hwanghakbeom Can you share the configuration how you pushed the logs to GCS ( Using HMAC and S3 output plugin)

**Reply by vg-meesho** [reply]
+1

**Reply by fancybear-dev** [reply]
https://docs.fluentbit.io/manual/data-pipeline/outputs/s3#usage-with-google-cloud
